stellar_sdk/soroban_server_async.py

```diff
@@ -235,6 +236,42 @@ async def send_transaction(
         )
         return await self._post(request, SendTransactionResponse)
 
+    async def poll_transaction(
+        self,
+        transaction_hash: str,
+        attempts: int = 30,
+        sleep_strategy: SleepStrategy = BasicSleepStrategy,
+    ) -> GetTransactionResponse:
+        """Poll for a particular transaction with certain parameters.
+
+        After submitting a transaction, clients can use this to poll for transaction completion and return a definitive state of success or failure.
+
+        :param transaction_hash: The hash of the transaction to poll for.
+        :type transaction_hash: str
+        :param attempts: The number of attempts to make before returning the last-seen status, defaults to 30.
+        :type attempts: int, optional
+        :param sleep_strategy: The amount of time to wait for between each attempt, defaults to 1 second between each attempt.
+        :type sleep_strategy: SleepStrategy, optional
+        :return: A :class:`GetTransactionResponse <stellar_sdk.soroban_rpc.GetTransactionResponse>` response object after a "found" response, (which may be success or failure) or the last response obtained after polling the maximum number of specified attempts.
+        :rtype: GetTransactionResponse
+        """
+        # positive and defined user value or default
+        max_attempts: int = DEFAULT_POLLING_ATTEMPTS if (attempts or 0) < 1 else attempts or DEFAULT_POLLING_ATTEMPTS
+
+        attempt: int = 0
+        resp = await self.get_transaction(transaction_hash=transaction_hash)
+
+        while attempt < max_attempts:
+            if (resp.status != GetTransactionStatus.NOT_FOUND):
+                return resp
+
+            attempt += 1
+            resp = await self.get_transaction(transaction_hash=transaction_hash)
+
+            time.sleep(sleep_strategy(attempt))
```

Comment:
```suggestion
            await asyncio.sleep(sleep_strategy(attempt))
```

See https://stackoverflow.com/questions/56729764/asyncio-sleep-vs-time-sleep

Reply:
That is a _really helpful_ explanation! Thank you for pointing it out for me